packages/hierarchy-builder/src/hierarchy-builder/HierarchyNode.ts

```diff
@@ -106,6 +106,8 @@ export interface BaseGroupingParams {
   hideIfNoSiblings?: boolean;
   /** Hiding option that determines whether to hide group nodes which have only one node as its children. */
   hideIfOneGroupedNode?: boolean;
+  /** Option which auto expands grouping nodes' children when it has single child or always. */
+  autoExpand?: "single-child" | "always";
```

Comment:
I think it would be worth creating a `type` for this string union. At the moment we don't have type safety everywhere, e.g.  in `getGroupingAutoExpandOptionsFromParentNode`.

Reply:
Added:
```
export interface BaseGroupingParams {
  /** Hiding option that determines whether to hide group nodes which have no siblings at the same hierarchy level. */
  hideIfNoSiblings?: boolean;
  /** Hiding option that determines whether to hide group nodes which have only one node as its children. */
  hideIfOneGroupedNode?: boolean;
  /** Option which auto expands grouping nodes' children when it has single child or always. */
  autoExpand?: AutoExpand;
}
/**
 * A [[BaseGroupingParams]] autoExpand option used to expand grouping nodes' children.
 * @beta
 */
export type AutoExpand = "single-child" | "always";
```